libr/core/cmd_debug.c

```diff
@@ -1994,8 +1932,10 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
 		bpi = r_bp_get_at (core->dbg->bp, addr);
 		if (bpi) {
 			//bp->enabled = !bp->enabled;
+			// XXX(jjd): this ^^ is what I would think toggling means...
 			r_bp_del (core->dbg->bp, addr);
 		} else {
+			// XXX(jjd): does t his need an address validity check??
 			bpi = r_debug_bp_add (core->dbg, addr, hwbp, NULL, 0);
```

Comment:
The goal for ```r_debug_bp_add``` is to add an R_API function to automate much of adding a breakpoint. Unfortunately "validAddress" uses "core" to determine validity. I'd guess this is the reason that function lives in cmd_debug.c instead of debug.c.